tests/db_functions/json/test_json_set.py

```diff
@@ -0,0 +1,403 @@
+import decimal
+import json
+
+from django.db import NotSupportedError, connection
+from django.db.models import F, JSONField, Value
+from django.db.models.functions import JSONObject
+from django.db.models.functions.json import JSONSet
+from django.test import TestCase, skipIfDBFeature, skipUnlessDBFeature
+
+from ..models import UserPreferences
+
+
+@skipUnlessDBFeature("supports_partial_json_update")
+class JSONSetTests(TestCase):
```

Comment:
I think many of the test methods here can be combined into a single method with `subTest`.